cranelift-codegen/src/verifier/mod.rs

```diff
@@ -1469,6 +1472,45 @@ impl<'a> Verifier<'a> {
         Ok(())
     }
 
+    fn typecheck_copy_nop(
+        &self,
+        inst: Inst,
+        errors: &mut VerifierErrors,
+    ) -> VerifierStepResult<()> {
+        if let InstructionData::Unary {
+            opcode: Opcode::CopyNop,
+            arg,
+        } = self.func.dfg[inst]
+        {
+            let dst_vals = self.func.dfg.inst_results(inst);
+            if dst_vals.len() != 1 {
+                return fatal!(errors, inst, "copy_nop must produce exactly one result");
+            }
+            let dst_val = dst_vals[0];
+            let src_type = self.func.dfg.value_type(arg);
+            let dst_type = self.func.dfg.value_type(dst_val);
```

Comment:
nit: since these two variables are used only once, can you replace their uses by their definitions?